### PR TITLE
fix: partially revert invalidate focus ring

### LINKED ISSUE #41839: [Bug]: SIGSEGV crash when maximizing window
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

30.0.0-alpha7

### What operating system are you using?

Ubuntu

### Operating System Version

20.04, running on x11

### What arch are you using?

x64

### Last Known Working Electron version

30.0.0-alpha6

### Expected Behavior

Maximizing/fullscreening the app does not sigfault.

### Actual Behavior

Maximizing/fullscreening the window crashes the app.

### Testcase Gist URL

This is visible in the default Fiddle template that opens with the app

### Additional Information

There is no `render-process-gone` or `child-process-gone` event called when it crashes.

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

Fixes #41839

We were seeing an inconsistent segfault on Ubuntu and several other Linux distros, and traced it back to this CL:https://chromium-review.googlesource.com/c/chromium/src/+/5344356. It appeats that the current upstream layout invalidation logic doesn't always invalidate a parent and child correctly. A label's bounds could change without being invalidated because its container can change its bounds.

We should handle this layout logic to better handle parent/child invalidation (see #41899, which is in progress) and upstream a fix if possible, but this PR should unblock developers wanting to upgrade to E31 and E30.

cc @ckerr @Kilian 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed an inconsistent crash on maximizing window and relayout in Ubuntu.

### CODE REVIEW

**Review by dsanders11 — APPROVED** [review]

**Review by ckerr — APPROVED** [review]
tldr: :+1: to everything @VerteDinde said in the description.

- I've confirmed this fixes the issue for me in local builds
- This is a reasonable way to unblock releases while we figure out the root cause
- Once we track down the root cause, it would be preferable to fix that and drop this patch if possible
